### ISSUE TITLE
Description property for the chart legend does not override the default aria-label

### ISSUE THREAD

**Issue by MarkMcAmhlaidh** [vega/vega-lite#9642]
### Current Behaviour
- Legend elements always use default aria-label values regardless of the description property being set.
- Custom accessibility labels, set with the description property are ignored.
- Localisation of legend text is not possible.

### Expected Behavior
- The description property should properly override default aria-label values.
https://vega.github.io/vega-lite/docs/legend.html#properties

### Impact 
- **Accessibility:** Users relying on screen readers cannot understand chart legends in their language
- **Localisation:** Cannot provide localised aria-labels international users

### Reproduction Steps 
- Create a chart with a legend
- Set the description property on the legend configuration
- Inspect the legend elements on your browser
- You will see that the default values are used instead of the custom description
```
{
  "$schema": "https://vega.github.io/schema/vega-lite/v6.json",
  "data": {"url": "data/barley.json"},
  "mark": "bar",
  "encoding": {
       "x": {"aggregate": "sum", "field": "yield"},
       "y": {"field": "variety"},
       "color": {
          "field": "site", 
          "legend": {
             "description": "test custom label",
          }
       }
  }
}
```

<img width="535" height="271" alt="Image" src="https://github.com/user-attachments/assets/65dcdbfe-d08f-44f9-b5b7-a9d7634c9f31" />

<img width="826" height="116" alt="Image" src="https://github.com/user-attachments/assets/a5bc4713-2b69-4121-bebb-8d8a2b85f27f" />

**Comment by domoritz** [comment]
Please add an example in the issue. Also, would you be willing to look into a fix?
